### PR TITLE
First integration

### PR DESCRIPTION
**Pull request by jeffkimble**
This is a wrapper for the netuitive docker agent to enable host data collection as a daemonset on every K8S node.

This is also enabled ( but not utilized ) to serve as a statsd endpoint which is why it includes a service.

This allows us to install metricly agent without adding it to AMI creation

### CODE REVIEW

File: helm/metricly-agent/templates/metricly-agent.daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+spec:
+  selector:
+    matchLabels: 
+      name: {{ .Chart.Name }}
+  template:
+    metadata:
+      labels:
+        name: {{ .Chart.Name }}
+        project: {{ .Chart.Name }}
+        name: {{ .Chart.Name }}
```
Comment (agianey) [inline helm/metricly-agent/templates/metricly-agent.daemonset.yaml]: Name is set twice

File: helm/metricly-agent/templates/metricly-agent.service.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    name: {{ .Chart.Name }}
+    app: {{ .Chart.Name }}
+spec:
+
+  type: {{ .Values.service.privateServiceType | quote }}
+  ports:
+    - name: http
+      port: {{ .Values.app.port }}
+  selector:
+    deploymentconfig: {{ .Chart.Name }}
```
Comment (agianey) [inline helm/metricly-agent/templates/metricly-agent.service.yaml]: Is this file needed? It doesn't seem to be bound to any deployment as the selector `deploymentconfig` doesn't match any deployment's labels

File: README.md [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
+
+
+To Build:
+  aladdin build
+
+
+To Start Locally:
+  aladdin start
+
+
```
Comment (agianey) [inline README.md]: Could add publish step for completeness

File: build/build.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+PROJECT_NAME=metricly-agent
+
+echo "Building ${PROJECT_NAME} docker image (~30 seconds)"
+
+BUILD_PATH="$(cd "$(dirname "$0")"; pwd)"
+PROJ_ROOT="$(cd "$BUILD_PATH/.." ; pwd)"
+PRINT_ONLY="${PRINT_ONLY:-false}"
+HASH="${HASH:-local}"
+ALL="${ALL:-false}"
```
Comment (agianey) [inline build/build.sh]: ALL and PRINT_ONLY not used

File: helm/metricly-agent/templates/metricly-agent.configMap.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+{{ if .Values.agent.dockerHostname }}
+# Do not set DOCKER_HOSTNAME to an empty value. 
+#  DOCKER_HOSTNAME: {{ .Values.agent.dockerHostname | quote }}
+{{ end }}
+  APIKEY: {{ .Values.agent.apikey | quote }}
+  USE_LOCAL_CONFIG: {{ .Values.agent.use_local_config | quote }}
+  LPRT: {{ .Values.agent.lprt | quote }}
+  FORWARD: {{ .Values.agent.forward | quote }}	
+  FIP: {{ .Values.agent.fip | quote}}
+  FPRT:	{{ .Values.agent.fprt | quote}}
```
Comment (agianey) [inline helm/metricly-agent/templates/metricly-agent.configMap.yaml]: Random indent

Comment (jeffkimble) [reply helm/metricly-agent/templates/metricly-agent.configMap.yaml]: didn't see it in the source doc..  deleted whitespace and replaced wtih space

File: helm/metricly-agent/templates/metricly-agent.daemonset.yaml (hunk 1 above)
Comment (jeffkimble) [reply helm/metricly-agent/templates/metricly-agent.daemonset.yaml]: Name is set as metadata and as label..

File: helm/metricly-agent/templates/metricly-agent.service.yaml (hunk 2 above)
Comment (jeffkimble) [reply helm/metricly-agent/templates/metricly-agent.service.yaml]: this is preliminary to enable statsd collection.. good catch on the selector.

File: helm/metricly-agent/templates/metricly-agent.daemonset.yaml (hunk 1 above)
Comment (agianey) [reply helm/metricly-agent/templates/metricly-agent.daemonset.yaml]: I meant it is set twice under metadata.labels
